pkg/secrets-store/provider_client.go

```diff
@@ -109,10 +110,15 @@ func (p *PluginClientBuilder) Get(ctx context.Context, provider string) (v1alpha
 		return nil, fmt.Errorf("%w: provider %q", ErrProviderNotFound, provider)
 	}
 
-	conn, err := grpc.DialContext(
-		ctx,
-		fmt.Sprintf("unix://%s/%s.sock", p.socketPath, provider),
+	conn, err := grpc.Dial(
+		fmt.Sprintf("%s/%s.sock", p.socketPath, provider),
 		grpc.WithInsecure(), // the interface is only secured through filesystem ACLs
+		grpc.WithContextDialer(func(ctx context.Context, target string) (net.Conn, error) {
+			// context.Background() is used instead of the request context
+			// because we want the client to live and re-connect if needed
+			// beyond the request's context duration.
+			return (&net.Dialer{}).DialContext(context.Background(), "unix", target)
```

Comment:
Another option is to alter `NewPluginClientBuilder()` to accept the server-process context instead of using Background(), but since the main thread will call `Cleanup()` on interrupt i'm not sure how necessary that is.

Reply:
actually the ctx in this function is actually the dialer func, and grpc.Dial() will wrap Background() for us.